pallets/settlement/src/lib.rs

```diff
@@ -2533,7 +2532,7 @@ impl<T: Config> Module<T> {
 
     /// Returns an instance of [`ExecuteInstructionInfo`].
     pub fn execute_instruction_info(instruction_id: &InstructionId) -> ExecuteInstructionInfo {
-        let caller_did = Identity::<T>::current_identity().unwrap_or(SettlementDID.as_id());
+        let caller_did = SettlementDID.as_id();
```

Comment:
Only used for RPC/Runtime API, so caller DID doesn't matter here.  If we really need it, then it should be a parameter.